tests/migrations/test_update_mlwh_with_legacy_samples.py

```diff
@@ -0,0 +1,162 @@
+def test_basic_usage(mongo_database, mlwh_connection):
+    config, mongo_db = mongo_database
+
+    start_datetime = datetime(2020, 5, 10, 15, 10)
+
+    # generate and insert sample rows into the mobgo database
+    test_samples = generate_example_samples(range(0, 6), start_datetime)
+    mongo_db.samples.insert_many(test_samples)
+
+    total_samples = mongo_db.samples.count_documents({})
+    assert total_samples == 6
+
+    # connect to the MySQL MLWH database
+    _, mysql_conn = mlwh_connection
+
+    # clear any existing rows in the lighthouse sample table
+    clear_mlwh_table(mysql_conn)
+
+    s_start_datetime = datetime.strftime(start_datetime, MONGO_DATETIME_FORMAT)
+    s_end_datetime = datetime.strftime(start_datetime + timedelta(days=3), MONGO_DATETIME_FORMAT)
+
+    # run the method to update the MLWH from the mongo database
+    update_mlwh_with_legacy_samples(config, s_start_datetime, s_end_datetime)
+
+    # query for selecting rows from MLWH (it was emptied before so select * is fine for this)
+    sql_query = f"SELECT * FROM unified_warehouse_test.{MLWH_TABLE_NAME} ORDER BY {FIELD_CREATED_AT} ASC"
+
+    try:
+        # run the query and fetch the results
+        cursor = mysql_conn.cursor()
+        cursor.execute(sql_query)
+        records = cursor.fetchall()
+
+        # check there are the expected number of rows in the MLWH (4?)
+        assert cursor.rowcount == 4
+
+        # check the plate barcodes are as expected
+        assert records[0][5] == "DN10000000"
+        assert records[3][5] == "DN10000003"
+
+    except Exception as e:
+        print("An exception occurred checking the mlwh table for rows inserted")
+        print_exception(e)
+    finally:
+        cursor.close()
+        mysql_conn.close()
+
+def test_when_no_rows_match_timestamp_range(mongo_database, mlwh_connection):
+    config, mongo_db = mongo_database
+
+    start_datetime = datetime(2020, 5, 10, 15, 10)
+
+    # generate and insert sample rows into the mobgo database
+    test_samples = generate_example_samples(range(0, 6), start_datetime)
+    mongo_db.samples.insert_many(test_samples)
+
+    total_samples = mongo_db.samples.count_documents({})
+    assert total_samples == 6
+
+    # connect to the MySQL MLWH database
+    _, mysql_conn = mlwh_connection
+
+    # clear any existing rows in the lighthouse sample table
+    clear_mlwh_table(mysql_conn)
+
+    # use timestamps such that no rows qualify
+    s_start_datetime = datetime.strftime(start_datetime + timedelta(days=6), MONGO_DATETIME_FORMAT)
+    s_end_datetime = datetime.strftime(start_datetime + timedelta(days=8), MONGO_DATETIME_FORMAT)
+
+    # run the method to update the MLWH from the mongo database
+    update_mlwh_with_legacy_samples(config, s_start_datetime, s_end_datetime)
+
+    # query for selecting rows from MLWH (it was emptied before so select * is fine for this)
+    sql_query = f"SELECT * FROM unified_warehouse_test.{MLWH_TABLE_NAME} ORDER BY {FIELD_CREATED_AT} ASC"
```

Comment:
Same comments as same line in previous test

Reply:
Changed